test/orange/orangeinp/detail/PolygonUtils.test.cc

```diff
@@ -149,6 +162,118 @@ TEST(PolygonUtilsTest, convex_self_intersect)
+ *   .                |          .
+ *   .                |         .
+ *   b________________a____g__f_________
+ *   |        |       |       |        |
+ *  -1       -0.5     0      0.5       1
+ */
+TEST(PolygonUtilsTest, filter_collinear_points)
+{
+    // Points locations, as labled above
+    Real2 a = {0, 0};
+    Real2 b = {-1, -1e-5};
+    Real2 c = {-0.9, -0.1};
+    Real2 d = {0.75, 1};
+    Real2 e = {0.75, 0.5};
+    Real2 f = {0.5, 1e-5};
+    Real2 g = {0.35, 1e-6};
+
+    using VecReal2 = std::vector<Real2>;
+    real_type tol = 0.01;
+
+    // no collinear points (b through f)
+    VecReal2 points{b, c, d, e, f};
+    VecReal2 exp = points;
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Point a is collinear, using a through f, and a comes first
+    points = {a, b, c, d, e, f};
+    exp = {b, c, d, e, f};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a is second
+    points = {f, a, b, c, d, e};
+    exp = {f, b, c, d, e};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a is third
+    points = {e, f, a, b, c, d};
+    exp = {e, f, b, c, d};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a is last
+    points = {b, c, d, e, f, a};
+    exp = {b, c, d, e, f};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a is second
+    points = {f, a, b, c, d, e};
+    exp = {f, b, c, d, e};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Points a and g are collinear, using a through g, and a comes first
+    points = {a, b, c, d, e, f, g};
+    exp = {b, c, d, e, f};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a comes second
+    points = {g, a, b, c, d, e, f};
+    exp = {b, c, d, e, f};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a comes third
+    points = {f, g, a, b, c, d, e};
+    exp = {f, b, c, d, e};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a comes second to last
+    points = {c, d, e, f, g, a, b};
+    exp = {c, d, e, f, b};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+
+    // Same, but a comes last
+    points = {b, c, d, e, f, g, a};
+    exp = {b, c, d, e, f};
+    expect_polygon_equal(exp, filter_collinear_points(points, tol));
+}
+
+/* Test pathological case consisting of a many-sided regular polygon with every
+ * point locally soft collinear with its neighbors due to a large tolerance.
+ */
+TEST(PolygonUtilsTest, filter_collinear_points_pathological)
+{
+    // Create a many-sided regular polygon by placing 20 equally spaced points
+    // on a circule of radius = 1, in clockwise order.
+    VecReal2 points(20);
+    real_type step = static_cast<real_type>(2 * constants::pi / points.size());
```

Comment:
At this point, the cast is redundant.
```suggestion
    real_type step = 2 * constants::pi / static_cast<real_type>(points.size());
```